src/DependencyInjection/HttplugExtension.php

```diff
@@ -508,13 +533,79 @@ private function configurePlugin(ContainerBuilder $container, $serviceId, $plugi
     {
         $pluginServiceId = $serviceId.'.plugin.'.$pluginName;
 
-        $definition = class_exists(ChildDefinition::class)
-            ? new ChildDefinition('httplug.plugin.'.$pluginName)
-            : new DefinitionDecorator('httplug.plugin.'.$pluginName);
+        $definition = $this->getChildDefinition('httplug.plugin.'.$pluginName);
 
         $this->configurePluginByName($pluginName, $definition, $pluginConfig, $container, $pluginServiceId);
         $container->setDefinition($pluginServiceId, $definition);
 
         return $pluginServiceId;
     }
+
+    private function configureVcrPlugin(ContainerBuilder $container, array $config, $prefix)
+    {
+        $recorder = $config['recorder'];
+        $recorderId = in_array($recorder, ['filesystem', 'in_memory']) ? 'httplug.plugin.vcr.recorder.'.$recorder : $recorder;
+        $namingStrategyId = $config['naming_strategy'];
+        $replayId = $prefix.'.replay';
+        $recordId = $prefix.'.record';
+
+        if ('filesystem' === $recorder) {
+            $recorderDefinition = $this->getChildDefinition('httplug.plugin.vcr.recorder.filesystem');
+            $recorderDefinition->replaceArgument(0, $config['fixtures_directory']);
+            $recorderId = $prefix.'.recorder';
+
+            $container->setDefinition($recorderId, $recorderDefinition);
+        }
+
+        if ('default' === $config['naming_strategy']) {
+            $namingStrategyId = $prefix.'.naming_strategy';
+            $namingStrategy = $this->getChildDefinition('httplug.plugin.vcr.naming_strategy.path');
+
+            if (!empty($config['naming_strategy_options'])) {
+                $namingStrategy->setArguments([$config['naming_strategy_options']]);
+            }
+
+            $container->setDefinition($namingStrategyId, $namingStrategy);
+        }
+
+        $arguments = [
+            new Reference($namingStrategyId),
+            new Reference($recorderId),
+        ];
+        $record = new Definition(RecordPlugin::class, $arguments);
+        $replay = new Definition(ReplayPlugin::class, $arguments);
+        $plugins = [];
+
+        switch ($config['mode']) {
+            case 'replay':
+                $container->setDefinition($replayId, $replay);
+                $plugins[] = $replayId;
+                break;
+            case 'replay_or_record':
+                $replay->setArgument(2, false);
+                $container->setDefinition($replayId, $replay);
+                $container->setDefinition($recordId, $record);
+                $plugins[] = $replayId;
+                $plugins[] = $recordId;
+                break;
+            case 'record':
+                $container->setDefinition($recordId, $record);
+                $plugins[] = $recordId;
+                break;
+        }
+
+        return $plugins;
+    }
+
+    /**
+     * @param string $parent the parent service id
+     *
+     * @return ChildDefinition
+     */
+    private function getChildDefinition($parent)
```

Comment:
can we name this `createChildDefinition`? i was trying to figure out why we can access a definition before the service file defining the vcr definitions has been loaded... until i read this method and realized it is a factory method (for the BC thingy).